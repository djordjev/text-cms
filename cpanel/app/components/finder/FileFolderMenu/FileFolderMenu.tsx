import { Link } from '@remix-run/react';
import { IconSquareX } from '@tabler/icons-react';
import { FC } from 'react';

export interface FileFolderMenuProps {
  path: string;
}

const FileFolderMenu: FC<FileFolderMenuProps> = (props) => {
  const { path } = props;

  return (
    <ul className="u-menu u-bg-base-200 u-w-2z u-rounded-box">
      <li>
        <Link
          className="hover:u-underline u-flex u-items-center"
          to={`?action=delete&path=${path}`}
        >
          <IconSquareX color="red" />
          Delete
        </Link>
      </li>
    </ul>
  );
};

export { FileFolderMenu };
