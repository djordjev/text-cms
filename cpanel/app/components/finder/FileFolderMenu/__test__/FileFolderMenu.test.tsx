import { createRemixStub } from '@remix-run/testing';
import { render, screen } from '@testing-library/react';

import { FileFolderMenu, FileFolderMenuProps } from '../FileFolderMenu';

test('ContextMenu', () => {
  const props: FileFolderMenuProps = { path: '/finder/something.txt' };

  const createComponent = () => {
    const RemixStub = createRemixStub([
      { path: '/', Component: () => <FileFolderMenu {...props} /> }
    ]);

    return <RemixStub />;
  };

  const { container } = render(createComponent());

  expect(screen.getByText('Delete')).toBeInTheDocument();

  expect(container.childNodes).toMatchSnapshot();
});
