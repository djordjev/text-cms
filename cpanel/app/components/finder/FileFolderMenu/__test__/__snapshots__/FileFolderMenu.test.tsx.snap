// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContextMenu 1`] = `
NodeList [
  <ul
    class="u-menu u-bg-base-200 u-w-2z u-rounded-box"
  >
    <li>
      <a
        class="hover:u-underline u-flex u-items-center"
        href="/?action=delete&path=/finder/something.txt"
      >
        <svg
          class="tabler-icon tabler-icon-square-x"
          fill="none"
          height="24"
          stroke="red"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 5a2 2 0 0 1 2 -2h14a2 2 0 0 1 2 2v14a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-14z"
          />
          <path
            d="M9 9l6 6m0 -6l-6 6"
          />
        </svg>
        Delete
      </a>
    </li>
  </ul>,
]
`;
